test/Infrastructure/LeanCode.ForceUpdate.Tests/ForceUpdateTests.cs

```diff
@@ -0,0 +1,110 @@
+using LeanCode.ForceUpdate.Contracts;
+using FluentAssertions;
+using Xunit;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+using LeanCode.CQRS.AspNetCore;
+using Microsoft.AspNetCore.Http;
+using LeanCode.CQRS.Execution;
+
+namespace LeanCode.ForceUpdate.Tests;
+
+public class ForceUpdateTests
+{
+    private const string MinimumRequiredVersion = "1.0";
+    private const string CurrentlySupportedVersion = "1.3";
+
+    private readonly ServiceProvider serviceProvider;
+
+    public ForceUpdateTests()
+    {
+        var services = new ServiceCollection();
+        services.AddCQRS(new(Array.Empty<Assembly>()), new(Array.Empty<Assembly>())).AddForceUpdate();
+        services.AddSingleton(
+            new IOSVersionsConfiguration(new Version(MinimumRequiredVersion), new Version(CurrentlySupportedVersion))
+        );
+        services.AddSingleton(
+            new AndroidVersionsConfiguration(
+                new Version(MinimumRequiredVersion),
+                new Version(CurrentlySupportedVersion)
+            )
+        );
+
+        this.serviceProvider = services.BuildServiceProvider();
+    }
+
+    [Fact]
+    public async Task Versions_correctly_returns_all_versions()
+    {
+        var handler = serviceProvider.GetRequiredService<IQueryHandler<Versions, List<VersionsDTO>>>();
+        var result = await handler.ExecuteAsync(new DefaultHttpContext(), new());
+
+        result
+            .Should()
+            .BeEquivalentTo(
+                new List<VersionsDTO>
+                {
+                    new VersionsDTO
+                    {
+                        Platform = PlatformDTO.IOS,
```

Comment:
We probably shouldn't rely on order of entries here, but making it a dictionary won't work without changing `Platform` to be `string`, or at least something serializable to `string` :weary:

I'd suggest requiring the client to pass `Platform` as query parameter for `Versions` but that puts in question separating that from `VersionSupport` query :weary:²